GitUI/HelperDialogs/FormResetAnotherBranch.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Windows;
+using GitCommands;
+using GitCommands.Git;
+using GitExtUtils.GitUI.Theming;
+using GitUIPluginInterfaces;
+using ResourceManager;
+
+namespace GitUI.HelperDialogs
+{
+    public partial class FormResetAnotherBranch : GitModuleForm
+    {
+        private IGitRef[] _localGitRefs;
+        private readonly GitRevision _revision;
+        private readonly TranslationString _localRefInvalid = new TranslationString("The entered value '{0}' is not the name of an existing local branch.");
+
+        public FormResetAnotherBranch(GitUICommands gitUiCommands, GitRevision revision)
+            : base(gitUiCommands)
+        {
+            _revision = revision;
+
+            InitializeComponent();
+            InitializeComplete();
+
+            labelResetBranchWarning.SetForeColorForBackColor();
+        }
+
+        private IGitRef[] GetLocalBranchesWithoutCurrent()
+        {
+            var currentBranch = Module.GetSelectedBranch();
+            var isDetachedHead = currentBranch == DetachedHeadParser.DetachedBranch;
+
+            return Module.GetRefs(false)
+                .Where(r => r.IsHead && (isDetachedHead || r.LocalName != currentBranch))
+                .ToArray();
+        }
+
+        private void FormResetAnotherBranch_Load(object sender, EventArgs e)
+        {
+            if (_revision == null)
+            {
+                throw new Exception(Strings.NoRevision);
+            }
```

Comment:
I'd rather do this check in the ctor and use the slightly more understandable `OperationNotSupportedException`.